DigitalLearningSolutions.Web/ViewModels/Common/ViewComponents/UnverifiedEmailListViewModel.cs

```diff
@@ -8,25 +8,41 @@ public class UnverifiedEmailListViewModel
         public readonly bool AtLeastOneCentreEmailIsUnverified;
         public readonly string? PrimaryEmailIfUnverified;
         public readonly bool PrimaryEmailIsVerified;
-        public readonly Dictionary<string, List<string>> UnverifiedCentreEmails;
+        public readonly Dictionary<string, List<string>> UnverifiedCentreEmailsDifferentFromPrimaryEmail;
 
         public UnverifiedEmailListViewModel(
             string? primaryEmailIfUnverified,
             Dictionary<string, List<string>> unverifiedCentreEmails
         )
         {
             PrimaryEmailIfUnverified = primaryEmailIfUnverified;
-            UnverifiedCentreEmails = unverifiedCentreEmails;
             PrimaryEmailIsVerified = string.IsNullOrWhiteSpace(primaryEmailIfUnverified);
             AtLeastOneCentreEmailIsUnverified = unverifiedCentreEmails.Any();
+
+            CentresWhereUnverifiedCentreEmailIsSameAsPrimaryEmail =
+                !PrimaryEmailIsVerified && unverifiedCentreEmails.ContainsKey(PrimaryEmailIfUnverified!)
+                    ? unverifiedCentreEmails[PrimaryEmailIfUnverified]
+                    : null;
+            PrimaryEmailIsUnverifiedAndTheSameAsAnUnverifiedCentreEmail =
+                !PrimaryEmailIsVerified && CentresWhereUnverifiedCentreEmailIsSameAsPrimaryEmail != null;
+            UnverifiedCentreEmailsDifferentFromPrimaryEmail =
+                GetUnverifiedCentreEmailsDifferentFromPrimaryEmail(unverifiedCentreEmails);
         }
 
-        public List<string>? CentresWhereUnverifiedCentreEmailIsSameAsPrimaryEmail =>
-            PrimaryEmailIfUnverified != null && UnverifiedCentreEmails.ContainsKey(PrimaryEmailIfUnverified)
-                ? UnverifiedCentreEmails[PrimaryEmailIfUnverified]
-                : null;
+        public List<string>? CentresWhereUnverifiedCentreEmailIsSameAsPrimaryEmail { get; set; }
+        public bool PrimaryEmailIsUnverifiedAndTheSameAsAnUnverifiedCentreEmail { get; set; }
+
+        private Dictionary<string, List<string>> GetUnverifiedCentreEmailsDifferentFromPrimaryEmail(
+            Dictionary<string, List<string>> unverifiedCentreEmails
+        )
+        {
+            if (PrimaryEmailIsUnverifiedAndTheSameAsAnUnverifiedCentreEmail)
+            {
+                unverifiedCentreEmails.Remove(PrimaryEmailIfUnverified!);
+                return unverifiedCentreEmails;
+            }
 
-        public bool PrimaryEmailIsUnverifiedAndTheSameAsAnUnverifiedCentreEmail => !PrimaryEmailIsVerified &&
-            CentresWhereUnverifiedCentreEmailIsSameAsPrimaryEmail != null;
+            return unverifiedCentreEmails;

```

Comment:
Oneline option: `unverifiedCentreEmails.Except(PrimaryEmailIfUnverified!)`

Reply:
Ah sorry, unverifiedCentreEmails is a dictionary and PrimaryEmailIfUnverified is a string, so this wouldn't work.